### PR TITLE
Feature/convert attribute to struct

### PR DESCRIPTION
**Pull request by daniel-flemming**
This change should avoid a ton of object creations and reduce GC runs significantly.

### CODE REVIEW

File: src/Api.Rest.Dtos/Data/AttributeDto.cs [hunk 1]
```diff
@@ -27,22 +27,16 @@ namespace Zeiss.PiWeb.Api.Rest.Dtos.Data
 	/// <remarks>This class is immutable.</remarks>
 	[Serializable]
 	[JsonConverter( typeof( AttributeConverter ) )]
-	public class AttributeDto : IEquatable<AttributeDto>
```
Comment (czthiele) [inline src/Api.Rest.Dtos/Data/AttributeDto.cs:30]: Looks very good. I have added some basic tests for this class in PR #150.

File: src/Api.Rest.Dtos/Data/AttributeDto.cs [hunk 2]
```diff
@@ -108,6 +102,27 @@ public string Value
 
 		#region methods
 
+		/// <summary>
+		/// Validate the raw value and only allows null, int, double, DateTime
+		/// and CatalogEntry as a RawValue
+		/// </summary>
+		private void ValidateRawValue()
```
Comment (czthiele) [inline src/Api.Rest.Dtos/Data/AttributeDto.cs:109]: Usage of this new method is missing

**Review by czthiele — APPROVED** [review]

File: src/Api.Rest.Dtos/Data/AttributeDto.cs (hunk 1 above)
Comment (daniel-flemming) [reply src/Api.Rest.Dtos/Data/AttributeDto.cs:30]: Perfect. Thank you :-)

File: src/Api.Rest.Dtos/Data/AttributeDto.cs (hunk 2 above)
Comment (daniel-flemming) [reply src/Api.Rest.Dtos/Data/AttributeDto.cs:110]: Fixed.

**Pull request merged**